hack/generate/csv.sh

```diff
@@ -14,6 +14,7 @@ serving="${registry}/knative-v$(metadata.get dependencies.serving):knative-servi
 client_version="$(metadata.get dependencies.cli)"
 kn_event="${registry_host}/knative/release-${client_version%.*}:client-plugin-event"
 rbac_proxy="registry.ci.openshift.org/origin/4.7:kube-rbac-proxy"
+kn_func_socat="quay.io/boson/alpine-socat:1.7.4.3-r1-non-root"
```

Comment:
I am a little confused. I thought this should be the productized socat image.

Reply:
We have 3 types of images:

- Upstream images, like this one
- Midstream images, build by Openshift CI (and new, upcoming, multi-arch ones, build by GH Actions - @mgencur have done them for eventing-kafka). Those images are built from midstream repos, and are updates after every PR that lands. This makes testing on midstream PRs reliable, as we are testing vs. midstream code, not upstream.
- Productized images - built from productization mirror repos

Here we have an upstream image. We should have here a midstream image. Also, that midstream image should be used in func midstream test suite. We don't have such image as of now.

I think we should use the upstream image for 1.25, but for 1.26 we should build a proper midstream image (probably by @mgencur GH action), and func team should adjust the code to use it in PR testing.

Reply:
> I am a little confused. I thought this should be the productized socat image.

Per Chris comment. This's a midstream hook, since we don't have anything better than upstream image right now. For the purpose of getting the references in. It should be fine. 

The final value of socat image will be injected though productization builds. Afaik, it's better for prod automation to have those image hooks precreated to replace the values in the building process.




> Maybe we should add this version of socat image to `project.yaml`?
> 
> On the other hand, if we plan to have this image build on midstream (via OpenShift CI or GH Actions), which we should, then it may not be sensible. @lance WDYT?

Yes, completely agreed. Nevertheless, I'd add `project.yaml` once we have midstream socat image in next release.  Then we can build it from metadata references, similarly to others.